youtube_dl/extractor/bbc.py

```diff
@@ -1074,6 +1086,85 @@ def _real_extract(self, url):
                     'subtitles': subtitles,
                 }
 
+        # simorgh-based playlist (see https://github.com/bbc/simorgh)
+        # JSON assigned to window.SIMORGH_DATA in a <script> element
+        simorgh_data = self._parse_json(
+            self._search_regex(
+                r'window\.SIMORGH_DATA\s*=\s*(\{[^<]+})\s*</',
+                webpage, 'simorgh playlist', default='{}'),
+            playlist_id, fatal=False)
+        # legacy media, video in promo object (eg, http://www.bbc.com/mundo/video_fotos/2015/06/150619_video_honduras_militares_hospitales_corrupcion_aw)
+        playlist = try_get(simorgh_data, lambda x: x['pageData']['promo']['media']['playlist']) or []
+        if playlist:
+            media = simorgh_data['pageData']['promo']
+            if media['media'].get('format') == 'video':
+                media.update(media['media'])
+                formats = []
+                keys = {'url', 'format', 'format_id', 'language', 'quality', 'tbr', 'resolution'}
+                for format in playlist:
+                    if not (format.get('url') and format.get('format')):
+                        continue
+                    bitrate = format.pop('bitrate')
+                    if bitrate:
+                        bitrate = re.sub(r'000\s*$', 'kbps', bitrate)
+                        format['tbr'] = parse_bitrate(bitrate)
```

Comment:
This looks like overkill. Why not just use `float_or_none`?

Reply:
Compare `unified_timestamp` vs `parse_iso8601`, though? Who knows what the BBC will send here? But if it ends in `000`, they probably mean kbps ...

Reply:
Then something like `float_or_none(bitrate, scale=1000) or parse_bitrate(bitrate)` should be used.
Current approach also won't work if `bitrate` is not evenly divisible by 1000.

Reply:
That's a good alternative, done.